### PR TITLE
Switch to `postcss-nesting` & update `postcss-mixins`

### PR DESCRIPTION
**Pull request by rdimascio**
### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by joesnellpdx** [comment]
@nicholasio @rdimascio 

can you confirm this will work with BrowserSync and hot reloading?
can you also confirm there are no regression when combined with https://github.com/10up/10up-toolkit/pull/112

This is an important update we should move forward with.

**Comment by Antonio-Laguna** [comment]
If there's anything I can help with here let me know, happy to dedicate some time.

That said, the drawback to this is that this will be a breaking change. Projects can't simply migrate from Sass nesting to Spec nesting. IMHO it's worth aligning with the spec as, someday, it will hopefully will just get removed as a PostCSS plugin and things will "just work™". 

@rdimascio I'm going to add some comments to the PR.

File: packages/toolkit/config/postcss.config.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ module.exports = ({ file, env }) => {
 		plugins: {
 			'postcss-import': {},
 			'postcss-mixins': {},
-			'postcss-nested': {},
+			'postcss-nesting': {},
```
Comment (Antonio-Laguna) [inline packages/toolkit/config/postcss.config.js:8]: Don't need to add this. `postcss-preset-env` handles this.

File: packages/toolkit/package.json [hunk 2]
```diff
@@ -58,13 +58,14 @@
...
-    "postcss": "^8.3.6",
+    "postcss": "^8.3.11",
     "postcss-css-variables": "^0.18.0",
     "postcss-editor-styles": "^0.3.0",
     "postcss-import": "^14.0.2",
     "postcss-loader": "^5.0.0",
-    "postcss-mixins": "^7.0.3",
+    "postcss-mixins": "^8.1.0",
     "postcss-nested": "^5.0.6",
+    "postcss-nesting": "^9.0.0",
```
Comment (Antonio-Laguna) [inline packages/toolkit/package.json:68]: Same, just get rid of this

File: packages/toolkit/package.json [hunk 3]
```diff
@@ -58,13 +58,14 @@
...
     "postcss-object-fit-images": "^1.1.2",
     "postcss-preset-env": "^6.7.0",
```
Comment (Antonio-Laguna) [inline packages/toolkit/package.json:70]: Needs to get updated to 7.0.0

**Comment by nicholasio** [comment]
> If there's anything I can help with here let me know, happy to dedicate some time.
> 
> That said, the drawback to this is that this will be a breaking change. Projects can't simply migrate from Sass nesting to Spec nesting. IMHO it's worth aligning with the spec as, someday, it will hopefully will just get removed as a PostCSS plugin and things will "just work™".
> 
> @rdimascio I'm going to add some comments to the PR.

@joesnellpdx Thoughts on switching to postcss-nesting?

**Comment by joesnellpdx** [comment]
> > If there's anything I can help with here let me know, happy to dedicate some time.
> > That said, the drawback to this is that this will be a breaking change. Projects can't simply migrate from Sass nesting to Spec nesting. IMHO it's worth aligning with the spec as, someday, it will hopefully will just get removed as a PostCSS plugin and things will "just work™".
> > @rdimascio I'm going to add some comments to the PR.
> 
> @joesnellpdx Thoughts on switching to postcss-nesting?

From what I am reading in the conversations, moving to `postcss-preset-env` and removing `postcss-nesting` and `post-css-nested` will resolve our issues... If I'm understanding @Antonio-Laguna 's comments correctly.

The MAIN thing for me, is whatever solution we choose it is tested thoroughly (build, start, watch - browsersync) and things work as expected.

We don't need to worry about scss as we typically don't have that on projects (IMO). 

If it is a breaking change from the current implementation for postcss, though, we need to move 10up-toolkit to 2.0.0.

I'm wondering if we should combine this MR with the cssnano update as they are connected and I want to be sure all this works together?: https://github.com/10up/10up-toolkit/pull/112

Once we validate all works as expected with this change, I approve moving forward. 

@rdimascio @nicholasio

**Comment by nicholasio** [comment]
@joesnellpdx Makes sense to me.

@rdimascio @Antonio-Laguna Let's move forward with postcss-preset-env latest release and postcss-nesting. Can either of you update this PR?

I can look into some sort of snapshot testing for the compiled CSS to ensure we're transforming CSS as expected.

**Comment by nicholasio** [comment]
I have included this in #114 which I'm also updating other deps that will be part of the new major version.

**Pull request closed without merging**